### PR TITLE
refactor: tests for container and cpc

### PR DESCRIPTION
**Pull request by aleyeeey**

### CODE REVIEW

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/inspector/JarInspector.java [hunk 1]
```diff
@@ -722,4 +722,14 @@ private String buildPurl(String namespace, String name, String version, String t
         return null;
     }
 
+    private String extractPackageName(String path) {
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/inspector/JarInspector.java]: fix artifact.deriveArtifactId

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/ApkPackageContributor.java [hunk 2]
```diff
@@ -131,9 +131,13 @@ public List<ComponentPatternData> contribute(File baseDir, String virtualRootPat
...
                             LOG.warn("No include patterns found for package: [{}-{}-{}]", packageName, version, architecture);
-                            processCollectedData(components, packageName, version, architecture, "**/*", virtualRoot.relativize(relativeAnchorFile).toString(), anchorChecksum, license);
+                            // FIXME: collect only package-specific folders
+                            // FIXME: check names of folder (distribution-specific)
+                            includePatterns.add("usr/share/doc/" + packageName + "/**/*");
+                            includePatterns.add("usr/share/licenses/" + packageName + "/**/*");
+                            includePatterns.add("usr/share/man/" + packageName + "/**/*");
+                            processCollectedData(components, packageName, version, architecture, includePatterns.toString(), virtualRoot.relativize(relativeAnchorFile).toString(), anchorChecksum, license);
                         }
                     }
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/ApkPackageContributor.java:140]: move process after if-else

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/DotNetComponentPatternContributor.java [hunk 3]
```diff
@@ -120,8 +120,7 @@ private void processProjectFile(List<ComponentPatternData> components, String re
                 addComponent(components, packageName, version, relativeAnchorPath, anchorChecksum);
             }
         } catch (Exception e) {
-            // FIXME: adjust to contributor logging and exception handling convention
-            LOG.warn("Failure processing project file", e);
+            LOG.warn("Could not process project file", e);
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/DotNetComponentPatternContributor.java]: add file path to output; end with .

File: libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/conan-001/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from conans import ConanFile, CMake
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/conan-001/conanfile.py]: Add header; add link add license; better: obfuscate; reduce to minimum required

File: libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/go-001/go.mod [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module github.com/skyrocknroll/go-mod-example
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/go-001/go.mod]: Add header; add link add license

File: libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/mix-001/mix.exs [hunk 6]
```diff
@@ -0,0 +1,253 @@
+defmodule Phoenix.MixProject do
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/mix-001/mix.exs]: Add header; add link add license; better: obfuscate; reduce to minimum required

File: libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/cocoapod-001/HornsAppCore.podspec [hunk 7]
```diff
@@ -0,0 +1,23 @@
+Pod::Spec.new do |spec|
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/test/resources/component-pattern-contributor/cocoapod-001/HornsAppCore.podspec]: Add header; add link add license; better: obfuscate; reduce to minimum required

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/HexComponentPatternContributor.java [hunk 8]
```diff
@@ -113,9 +113,10 @@ public List<ComponentPatternData> contribute(File baseDir, String virtualRootPat
                         cpd.set(ComponentPatternData.Attribute.COMPONENT_PART, dependency + "-" + depVersion);
                         cpd.set(ComponentPatternData.Attribute.VERSION_ANCHOR, relativeAnchorPath);
                         cpd.set(ComponentPatternData.Attribute.VERSION_ANCHOR_CHECKSUM, anchorChecksum);
-                        cpd.set(ComponentPatternData.Attribute.INCLUDE_PATTERN, "**/*");
+                        cpd.set(ComponentPatternData.Attribute.INCLUDE_PATTERN, dependency + "/**/*");
```
Comment (karsten-klein) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/contributors/HexComponentPatternContributor.java]: Check

**Review by karsten-klein — COMMENTED** [review]
How does this relate to the other PR; let's review / merge as a whole. Changes in production code must be reviewed with due care.

**Comment by aleyeeey** [comment]
@karsten-klein this can be closed, since this is outdated.

**Pull request closed without merging**